public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx

```diff
@@ -91,6 +94,139 @@ describe('Role view - tenant panel', () => {
+        .find('[data-test-subj="tenant-permission-container"]')
+        .first()
+        .getElement();
+      const component = shallow(prompt);
+      const columns = component.prop<
+        Array<EuiTableFieldDataColumnType<RoleTenantPermissionDetail>>
+      >('columns');
+      const viewDashboardRenderer = columns[3].render as (tenant: string) => JSX.Element;
+      const Container = (props: { tenant: string }) => viewDashboardRenderer(props.tenant);
+      const result = shallow(<Container tenant={'tenant1'} />);
+      result.find('[data-test-subj="view-dashboard"]').simulate('click');
+      expect(mockTenantUtils.selectTenant).toHaveBeenCalled();
+    });
+
+    it('click on View Visualizations link', () => {
+      const tenantPermissions = [{ tenant_patterns: ['tenant1'], permissionType: 'dummy' }];
+      const wrapper = mount(
+        <TenantsPanel
+          roleName={sampleRoleName}
+          tenantPermissions={tenantPermissions}
+          errorFlag={false}
+          coreStart={mockCoreStart as any}
+          loading={false}
+          isReserved={true}
+        />
+      );
+
+      const prompt = wrapper
+        .find('[data-test-subj="tenant-permission-container"]')
+        .first()
+        .getElement();
+      const component = shallow(prompt);
+      const columns = component.prop<
+        Array<EuiTableFieldDataColumnType<RoleTenantPermissionDetail>>
+      >('columns');
+      const viewVisualizationRenderer = columns[4].render as (tenant: string) => JSX.Element;
+      const Container = (props: { tenant: string }) => viewVisualizationRenderer(props.tenant);
+      const result = shallow(<Container tenant={'tenant1'} />);
+      result.find('[data-test-subj="view-visualizations"]').simulate('click');
+      expect(mockTenantUtils.selectTenant).toHaveBeenCalled();
+    });
+
+    it('render view dashboard column when tenant is invalid', () => {
+      const tenantPermissions = [{ tenant_patterns: ['*'], permissionType: 'dummy' }];
+      const wrapper = mount(
+        <TenantsPanel
+          roleName={sampleRoleName}
+          tenantPermissions={tenantPermissions}
+          errorFlag={false}
+          coreStart={mockCoreStart as any}
+          loading={false}
+          isReserved={true}
+        />
+      );
+
+      const prompt = wrapper
+        .find('[data-test-subj="tenant-permission-container"]')
+        .first()
+        .getElement();
+      const component = shallow(prompt);
+      const columns = component.prop<
+        Array<EuiTableFieldDataColumnType<RoleTenantPermissionDetail>>
+      >('columns');
+      const viewDashboardRenderer = columns[3].render as (tenant: string) => JSX.Element;
+      const Container = (props: { tenant: string }) => viewDashboardRenderer(props.tenant);
+      const result = shallow(<Container tenant={RoleViewTenantInvalidText} />);
+      expect(result).toMatchSnapshot();
+    });
+
+    it('render view visualization column when tenant is invalid', () => {
```

Comment:
What does it mean by 'tenant is invalid'?

Reply:
It mean that when there are multiple tenants or tenant pattern as tenant permission.

Reply:
I'll change the text. It's confusing.